agent/src/main.rs

```diff
@@ -57,6 +55,22 @@ async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>
         run_metrics_server().await.unwrap();
     }));
 
+    let discovery_handler_map = Arc::new(Mutex::new(HashMap::new()));
+    let discovery_handler_map_clone = discovery_handler_map.clone();
+    let (new_discovery_handler_sender, _): (
+        broadcast::Sender<String>,
+        broadcast::Receiver<String>,
+    ) = broadcast::channel(4);
```

Comment:
4 looks kind of arbitrary, should we make it into a named constant with possibly some comment?

Reply:
Having a constant to explain it makes sense. However, i am not sure how to make the channel size non-arbitrary. 
The size defines the number of messages that are allowed to be queued in the `new_discovery_handler` channel (of which `AgentRegistration` and `DiscoveryOperators` have the sending and receiving ends, respectively) before the oldest message is dropped. A message will not be removed from the channel until all `DiscoveryOperators` have received it. There is a DiscoveryOperator per configuration, so teh more configurations, the more receivers. if you had 10 configurations applied and then 6 Discovery Handlers all registered at once I could see a [`RecvError::Lagged` error thrown](https://docs.rs/tokio/0.2.6/tokio/sync/broadcast/index.html#lagging). I am not sure what size to choose. Maybe it should be configurable?

Reply:
added in fa4eba408f57f2

Reply:
How do you keep track of how many DiscoveryOperators have received the message? What if DiscoveryOperator is added/removed mid-process?